### ISSUE TITLE
IDisposableOutput needed or i have to cast

### ISSUE THREAD

**Issue by ICACLE** [icarus-consulting/Yaapii.Atoms#72]
### Bug Report or Feature Request (mark with an x)
- [ ] bug report -> please search issues before submitting
- [x] feature request

### Expected Behavior

If i write a Decorator, which implements to Interfaces, i want to get in my ctor one interface of this type.
Example:
public seald class OutputOf : IOutput, IDisposable

My Decorator AppendTo have to take one interface inherit from both interfaces.
public interface IDisposableOutput : IOutput, IDisposable would solve my problems.

**Comment by Meerownymous** [comment]
I don‘t know if I understand your problem correctly, let me outline what I understood:

- we have the convention that we do not implement more than one interface
- your problem is: you are writing a IOutput decorator. This decorator encapsulates another IOutput or a stream. The stream needs to be disposable.
- to achieve this, you are forced to implement the Idispisable interface

Correct?

For this case, I think you do not need a new interface, I suggest that AppendTo implements two interfaces, IOutput and IDisposable. This breaks our rule, but we cannot change this because the .Net framework implements it this way. The other output classes are doing it the same way.

This includes the necessity to cast, of course, inside of the Dispose() method...

Does this help?

**Comment by ICACLE** [comment]
No, if we break it already, i think will have less pain if we break the 'a interface should not inherit from other interfaces'. Thin or class implements only one interface and we will be able to implent real decorators

**Comment by Meerownymous** [comment]
For me, I think seeing which interfaces are implemented with one click is more transparent than hiding it one level and introducing a new one for only this purpose is also a high cost..
Yep, lets discuss tomorrow!

**Comment by ICACLE** [comment]
We agreed to cast the IInput/IOutput to IDispsable and call the Dispose() Methode of the Parent

**Comment by ICACLE** [comment]
Done
